nucypher/policy/conditions/jwt.py

```diff
@@ -0,0 +1,168 @@
+        # self.issued_window = issued_window
+
+        self.logger = Logger(__name__)
+
+        super().__init__()
+
+    def execute(self, **context) -> Any:
+
+        jwt_token = resolve_any_context_variables(self.jwt_token, **context)
+
+        # header = jwt.get_unverified_header(self.jwt_token)
+        # algorithm = header['alg']
+        try:
+            payload = jwt.decode(
+                jwt_token, self.public_key, algorithms=self._valid_jwt_algorithms
+            )
+        except jwt.exceptions.InvalidAlgorithmError:
+            raise  # TODO: raise something specific
+        except jwt.exceptions.DecodeError:
+            raise
+
+        return payload
+
+
+class JWTCondition(ExecutionCallAccessControlCondition):
+    """
+    A JWT condition can be satisfied by presenting a valid JWT token, which not only is
+    required to be cryptographically verifiable, but also must fulfill certain additional
+    restrictions defined in the condition.
+    """
+
+    EXECUTION_CALL_TYPE = JWTVerificationCall
+    CONDITION_TYPE = ConditionType.JWT.value
+
+    class Schema(
+        ExecutionCallAccessControlCondition.Schema, JWTVerificationCall.Schema
+    ):
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.JWT.value), required=True
+        )
+        # jwt_token = fields.Str(required=True)  # TODO: validate jwt encoded format
+        # public_key = fields.Str(required=True)  # required? maybe a valid PK certificate passed by requester?
+        # expected_issuer = fields.Str(required=False)
+        # subject = fields.Str(required=False)
+        # expiration_window = fields.Int(
+        #     strict=True, required=False, validate=validate.Range(min=0), allow_none=True
+        # )
+        # issued_window = fields.Int(
+        #     strict=True, required=False, validate=validate.Range(min=0), allow_none=True
+        # )
+        # todo: kid (https://www.rfc-editor.org/rfc/rfc7515#section-4.1.4), x5u, etc
+
+        @post_load
+        def make(self, data, **kwargs):
+            return JWTCondition(**data)
+
+    def __init__(
+        self,
+        condition_type: str = ConditionType.JWT.value,
+        name: Optional[str] = None,
+        jwt_token: Optional[str] = None,
+        public_key: Optional[str] = None,
+        expected_issuer: Optional[str] = None,
+        # subject: Optional[str] = None,
+        # expiration_window: Optional[int] = None,
+        # issued_window: Optional[int] = None,
+    ):
+        super().__init__(
+            jwt_token=jwt_token,
+            public_key=public_key,
+            expected_issuer=expected_issuer,
+            # subject=subject,
+            # expiration=expiration_window,
+            # issued_window=issued_window,
+            condition_type=condition_type,
+            name=name,
+            return_value_test=ReturnValueTest(
+                comparator="==", value=True
+            ),  # TODO: Workaround for now
```

Comment:
Not necessarily the RTV evaluation logic itself, but the fact that RTVs are mandatory for ExecutionCalls is what I wanted to challenge, particularly for the case of execution calls that directly return a boolean value (this can also happen for offchain calls, for example). Maybe it's a very niche case, so I'm fine with RTVs even for this case.

Reply:
Just to clarify one thing. 

An `ExecutionCall` does not have an RTV but `ExecutionCallAccessControlCondition` does i.e. the wrapping of the call into a condition. Remember, a call is execution, a condition is value checking. 

If the evaluation of the underlying execution call is always a boolean as opposed to some value that you need to check, then `JWTCondition` can simply extend `AccessControlCondition` instead and still use the execution call internally. Alternatively, we could:
- add a further abstraction e.g. split the `ExecutionCallAccessControlCondtion` into two, one that uses RTV (underlying call returns a value that needs to be compared) and another that doesn't (the underlying call returns a boolean) 
OR 
- make `return_value_test` optional for the `ExecutionCallAccessControlCondition` (although this may involve needing to override the schema when RTV is required)...need to think more on this.

Side Note: Originally, the thought was that `ExecutionCall`s could be serialized and used on its own without a condition hence they had their own schema, and we ended up not doing that for SequentialConditions, but kept the absraction. One thought I have had more recently is that `ExecutionCall` should not define a Schema at all and just be functional; then subclassing of calls is easier, and conditions can still use them. This is coming up in the Non-Evm JSON RPC work...happy to chat more about it in voice.